### PR TITLE
Improve code coverage and integrate with SonarCloud

### PR DESCRIPTION
**Pull request by splitfeed**

### CODE REVIEW

**Comment by jepp3** [comment]
Nice to see that we will improve our coverage!!! :) :+1:

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+
+# must be unique in a given SonarQube instance
+sonar.projectKey=oss-reactive-wizard
```
Comment (jepp3) [inline sonar-project.properties]: will be very interesting :) :+1:

File: dbmigrate/src/test/java/se/fortnox/reactivewizard/dbmigrate/LiquibaseAutoBindModuleTest.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import static org.fest.assertions.Assertions.assertThat;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class LiquibaseAutoBindModuleTest {
+    @Test
+    public void testOnlyRunForDbArgument() throws LiquibaseException {
```
[the remark this answers is not in the document]
Comment (splitfeed) [reply dbmigrate/src/test/java/se/fortnox/reactivewizard/dbmigrate/LiquibaseAutoBindModuleTest.java]: 👍

**Comment by splitfeed** [comment]
Sonarcloud will not be functional until merged and run on master once, but it looks like it should work once merged :)

File: dbmigrate/src/main/java/se/fortnox/reactivewizard/dbmigrate/LiquibaseAutoBindModule.java [hunk 3]
```diff
@@ -1,33 +1,31 @@
...
-    private static Logger LOG = LoggerFactory.getLogger(LiquibaseAutoBindModule.class);
-    private final LiquibaseConfig liquibaseConfig;
-    private final String startCommand;
+    private final String           startCommand;
+    private final LiquibaseMigrate liquibaseMigrate;
 
     @Inject
-    public LiquibaseAutoBindModule(ConfigFactory confFactory, @Named("args") String[] args) {
-        liquibaseConfig = confFactory.get(LiquibaseConfig.class);
+    public LiquibaseAutoBindModule(@Named("args") String[] args, LiquibaseMigrate liquibaseMigrate) {
```
Comment (JonasHallFnx) [inline dbmigrate/src/main/java/se/fortnox/reactivewizard/dbmigrate/LiquibaseAutoBindModule.java]: When AutoBindMoudles are run, we are in the first of two Guice iterations. At that time, https://github.com/FortnoxAB/reactive-wizard/blob/master/config/src/main/java/se/fortnox/reactivewizard/config/ConfigAutoBindModule.java has not yet bound all Config-annotated classes to it's implementation, which means that you will always get an empty implementation here. Tricky.

Comment (splitfeed) [reply dbmigrate/src/main/java/se/fortnox/reactivewizard/dbmigrate/LiquibaseAutoBindModule.java]: Oh, I missed that. Any idea on how one could make the LiquibaseMigrate class mockable?

Comment (JonasHallFnx) [reply dbmigrate/src/main/java/se/fortnox/reactivewizard/dbmigrate/LiquibaseAutoBindModule.java]: You can inject a LiquibaseMigrateProvider here and let that class get a ConfigFactory to it's constructor. The ConfigFactory is bound at this time.

File: dbmigrate/src/test/java/se/fortnox/reactivewizard/dbmigrate/LiquibaseAutoBindModuleTest.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+        verify(liquibaseMigrateMock).drop();
+        verify(liquibaseMigrateMock).run();
+        verify(liquibaseMigrateMock, never()).exit();
+    }
+
+
+
+    @Test
+    public void shouldNotRunLiquibaseWhenArgumentsAreNull() throws LiquibaseException {
```
[the remark this answers is not in the document]
Comment (splitfeed) [reply dbmigrate/src/test/java/se/fortnox/reactivewizard/dbmigrate/LiquibaseAutoBindModuleTest.java]: You're right, I must have mixed up my tests :)

File: dbmigrate/src/main/java/se/fortnox/reactivewizard/dbmigrate/LiquibaseAutoBindModule.java (hunk 3 above)
Comment (splitfeed) [reply dbmigrate/src/main/java/se/fortnox/reactivewizard/dbmigrate/LiquibaseAutoBindModule.java]: That seems to have worked fine 👍

**Pull request merged**